app/forms/hub/take_action_form.rb

```diff
@@ -1,10 +1,28 @@
 module Hub
   class TakeActionForm < Form
-    attr_accessor :tax_returns, :locale, :message_body, :contact_method, :internal_note_body
+    include MessageSending
+    attr_accessor :tax_return,
+                  :tax_return_id, 
+                  :tax_returns, 
```

Comment:
I wonder if you need all 3 - tax_return, tax_returns (plural), tax_return_id (id). Given that you have your own attribute accessor for tax_return `def tax_return` below, you may be able to remove all three.

Non-blocking.

Reply:
i think i still need tax_return_id like that in order to grab tax_return_id the way I do in the tax_return method, and then tax_returns is used to set up the list of tax returns in the dropdowns, so unfortunately i think i do need all three!